src/main/java/com/example/demo/dao/GradeLocalRepository.java

```diff
@@ -0,0 +1,73 @@
+package com.example.demo.dao;
+
+import com.example.demo.dao.entity.Grade;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Repository
+public class GradeLocalRepository implements GradeRepository {
+    private Map<Long, List<Grade>> gradeRegister;
+
+    public GradeLocalRepository() {
+        this.gradeRegister = new HashMap<>();
+    }
+
+    @Override
+    public List<Grade> getAllGrades() {
+        List<Grade> allGradesList = new ArrayList<>();
+        gradeRegister.values().stream()
+                .forEach(list -> allGradesList.addAll(list));
+        return allGradesList;
+    }
+
+    @Override
+    public List<Grade> getStudentGrades(long studentId) {
+        List<Grade> studentGradeList;
+        studentGradeList = gradeRegister.getOrDefault(studentId, List.of());
```

Comment:
```
return gradeRegister.getOrDefault(studentId, List.of());`
```
IntelliJ tego nie podpowiedział? 🤔 

`List.of` jest bodajże z Javy 9 (dobrze że jesteś świadomy że coś takiego istnieje), ale na 99% tworzy nową listę, a `Collections.emptyList()` tego nie robi

https://stackoverflow.com/questions/39400238/list-of-or-collections-emptylist